packages/caliper-core/lib/prometheus/prometheus-query-client.js

```diff
@@ -0,0 +1,166 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const url = require('url');
+const http = require('http');
+const https = require('https');
+
+const Logger = require('../utils/caliper-utils').getLogger('prometheus-query-client');
+
+const NULL = '/api/v1/';
+const RANGE = '/api/v1/query_range';
+const SINGLE = '/api/v1/query';
+
+/**
+ * PrometheusQueryClient for use with querying the Prometheus server to retrieve metrics
+ */
+class PrometheusQueryClient {
+
+    /**
+     * Constructor for client
+     * @param {String} prometheusUrl the url of the Prometheus gateway
+     */
+    constructor(prometheusUrl) {
+        this.prometheusUrl = prometheusUrl;
+    }
+
+    /**
+     * Configure the target for the push
+     * @param {String} type query type
+     * @param {String} query the unique query append
+     */
+    async configureTarget(type, query) {
+        const target = url.resolve(this.prometheusUrl, type + query);
+        this.requestParams = url.parse(target);
```

Comment:
what if queries (due to late answers) overlap? The `this.X` variables will be overwritten. 
The options of a query could be an independent object, passed around, instead of saved into an instance variable.

Reply:
good spot, this could be converted to a `retrieveTargetParameters` method that is passed through 👍